app/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { readJSON } from "fs-extra"
+import { PackageJson } from "gatsby"
+import path from "path"
+export async function hasGatsbyInstalled(root: string): Promise<boolean> {
+  try {
+    const packageJson = await readJSON(
+      path.join(root, `node_modules`, `gatsby`, `package.json`)
+    )
```

Comment:
Module resolution is expensive, and we don't need it here because we require gatsby to be a direct dependency. The lookup needs to be scoped to the particular site (it's not the current site)